RickAndMorty-iOS/Views/Locations/LocationDetailsViewController.swift

```diff
@@ -0,0 +1,179 @@
+//
+//  LocationDetailsViewController.swift
+//  RickAndMorty-iOS
+//
+//  Created by Calvin Pak on 2023-03-14.
+//
+
+import UIKit
+import Combine
+
+class LocationDetailsViewController: UIViewController {
+
+    enum LocationDetailsSection: Int, CaseIterable {
+        case info
+        case residents
+        case emptyInfo
+        case emptyResidents
+    }
+
+    weak var coordinator: MainCoordinator?
+    let locationDetailsView = LocationDetailsView()
+    let viewModel = LocationDetailsViewModel()
+    var locationId: String
+
+    typealias DataSource = UICollectionViewDiffableDataSource<LocationDetailsSection, AnyHashable>
+    typealias Snapshot = NSDiffableDataSourceSnapshot<LocationDetailsSection, AnyHashable>
+    private var dataSource: DataSource!
+    private var cancellables = Set<AnyCancellable>()
+    var snapshot = Snapshot()
```

Comment:
same for here. variables should be private where it is possible. 
Coordinator should be a part of ViewController